source/Octopus.Tentacle/Communications/HalibutInitializer.cs

```diff
@@ -100,7 +104,34 @@ void AddPollingEndpoints()
 
                 var halibutTimeoutsAndLimits = new HalibutTimeoutsAndLimits();
                 var serviceEndPoint = new ServiceEndPoint(pollingEndPoint.Address, pollingEndPoint.Thumbprint, halibutProxy, halibutTimeoutsAndLimits);
-                halibut.Poll(new Uri(pollingEndPoint.SubscriptionId), serviceEndPoint, CancellationToken.None);
+
+                //Open multiple polling connections if the env var is set to a non-zero/negative number
+                var connectionCount = 1u;
+                if (uint.TryParse(Environment.GetEnvironmentVariable(EnvironmentVariables.TentaclePollingConnectionCount), out var count))
+                {
+                    log.InfoFormat("Requested polling connection count: {0}", count);
+                    connectionCount = count;
+                }
+
+                //Coerce the requested value as it might be outside our max & min
+                switch (connectionCount)
+                {
+                    case > MaximumPollingConnectionCount:
+                        log.InfoFormat("Maximum polling connection count: {0}", MaximumPollingConnectionCount);
+                        connectionCount = 8;
+                        break;
+                    case 0:
+                        log.InfoFormat("Minimum polling connection count: {0}", 1);
+                        connectionCount = 1;
+                        break;
+                }
```

Comment:
Nit: it might be worth breaking this code out into it's own method.

Reply:
Done in [025f3ff](https://github.com/OctopusDeploy/OctopusTentacle/pull/881/commits/025f3ffed6571f06587c2cac2e8956c41e737aff)